doc/TockBinaryFormat.md

```diff
@@ -457,6 +519,82 @@ data between kernel and userspace and the the system call numbers.
 +-------------+-------------+---------------------------+
 ```
 
+#### `9` Program
+
+A Program Header is an extended form of the Main Header. It adds two
+fields, `binary_end_offset` and `version`. The `binary_end_offset` field
+allows the kernel to identify where in the TBF object the application
+binary ends. The gap between the end of the application binary and
+the end of the TBF object can contain footers. 
+
+```
+0             2             4             6             8
++-------------+-------------+---------------------------+
+| Type (9)    | Length (20) | init_offset               |
++-------------+-------------+---------------------------+
+| protected_size            | min_ram_size              |
++---------------------------+---------------------------+
+| binary_end_offset         | version                   |
++---------------------------+---------------------------+
+```
+
+  * `init_offset` the offset in bytes from the beginning of binary payload
+    (i.e. the actual application binary) that contains the first instruction to
+    execute (typically the `_start` symbol).
+  * `protected_size` the size of the protected region in bytes. Processes do not
+    have write access to the protected region. TBF headers are contained in the
+    protected region.
+  * `minimum_ram_size` the minimum amount of memory, in bytes, the process
+    needs.
```

Comment:
This should say what the kernel does if it does not have enough RAM left

Reply:
Isn't that a kernel policy, not a TBF specification?

Reply:
Yes, this is a kernel policy question. This document is just about the TBF format, we don't want to be specifying kernel behavior here.